src/main/java/seedu/address/logic/commands/RemoveTagCommand.java

```diff
@@ -21,36 +22,51 @@ public class RemoveTagCommand extends TagCommandGroup {
     public static final String COMMAND_SPECIFIER = "remove";
     public static final String COMMAND_WORD = COMMAND_GROUP + " " + COMMAND_SPECIFIER;
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Remove tags like this. "
-            + "Parameters: "
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Remove tags of the person identified "
+            + "by the index in the list or the target person if no index provided.\n"
+            + "Parameters: [INDEX] "
             + "[TAG1] [TAG2] ... \n"
             + "Example: " + COMMAND_WORD + " 1 friend";
 
     public static final String MESSAGE_SUCCESS = "Tags removed from user: %1$s";
     public static final String MESSAGE_TAGS_NOT_FOUND = "Tags do not exist: %1$s";
     public static final String MESSAGE_TAGS_NOT_BELONG_TO_USER = "Tags do not belong to user: %1$s";
 
-    private final Index index;
+    private final Optional<Index> index;
     private final Set<Tag> tagsToRemove;
 
     /**
+     * @param index of the person in the filtered person list to tag
      * @param tagsToRemove to tag the person with
      */
     public RemoveTagCommand(Index index, Set<Tag> tagsToRemove) {
         requireNonNull(index);
         requireNonNull(tagsToRemove);
-        this.index = index;
+        this.index = Optional.of(index);
+        this.tagsToRemove = tagsToRemove;
+    }
+
+    /**
+     * @param tagsToRemove to tag the person with
+     */
+    public RemoveTagCommand(Set<Tag> tagsToRemove) {
+        requireNonNull(tagsToRemove);
+        this.index = Optional.empty();
         this.tagsToRemove = tagsToRemove;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
         List<Person> lastShownList = model.getFilteredPersonList();
-        if (index.getZeroBased() >= lastShownList.size()) {
+        if (index.isPresent() && index.get().getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        } else if (!index.isPresent() && !model.hasTargetPerson()) {
+            throw new CommandException(Messages.MESSAGE_NO_TARGET_PERSON);
```

Comment:
Since you are already using optional, you can make use of methods of the optional class to improve abstraction.
```suggestion
Index personIndex = index.orElseThrow(() -> new CommandException(Messages.MESSAGE_NO_TARGET_PERSON));
if (personIndex.getZeroBased() => lastShownList.size()) {
    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
}
```